### PR TITLE
LPS-117871 Remove useMemo to always rerender the fields and the mask be applied to all

### PR DESCRIPTION
**Pull request by alinedoleron**

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
@alinedoleron, in the next pull, please remember to add the Jira ticket link. 😉

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -154,7 +152,7 @@ const DatePicker = ({
 			});
 			maskInstance.current.update(inputRef.current.value);
 		}
-	}, [inputMask, dateMask, inputRef]);
+	}, [inputMask, dateMask]);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:155]: @alinedoleron, why did you remove the `inputRef` from the `useEffect` array of dependencies? Won't it generate any side effect?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 2]
```diff
@@ -79,15 +79,13 @@ const getDelimiter = (dateFormat) => {
 };
 
 const useDateFormat = () => {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:81]: If we're removing the `useMemo` hook, then the function `useDateFormat` isn't a custom hook anymore, more details [here](https://reactjs.org/docs/hooks-custom.html), and, ideally, we should remove the prefix `use` and name this function as `getDateFormat`, for instance.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js (hunk 1 above)
Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:155]: Hi @natocesarrego, I removed because inputRef doesn't change. But I can put it back, anyway.

**Comment by alinedoleron** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by alinedoleron** [comment]
nw pr: https://github.com/natocesarrego/liferay-portal/pull/738
